modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/showSidebar.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {render} from 'frontend-js-react-web';
+
+import SidebarPanel from '../SidebarPanel';
+import SidebarPanelInfoView from '../components/SidebarPanelInfoView';
+import SidebarPanelMetricsView from '../components/SidebarPanelMetricsView';
+
+const actions = {
+	showInfo(fetchURL, portletNamespace) {
+		showSidebar({View: SidebarPanelInfoView, fetchURL, portletNamespace});
+	},
+	showMetrics(fetchURL, portletNamespace) {
+		showSidebar({
+			View: SidebarPanelMetricsView,
+			fetchURL,
+			portletNamespace,
+		});
+	},
+};
+
+const showSidebar = ({View, fetchURL, portletNamespace}) => {
+	const sidebarPanel = Liferay.component(`${portletNamespace}sidebar`);
+
+	if (!sidebarPanel) {
+		const container = document.body.appendChild(
+			document.createElement('div')
+		);
+
+		render(
+			SidebarPanel,
+			{
+				fetchURL,
+				ref: (element) => {
+					Liferay.component(`${portletNamespace}sidebar`, element);
+				},
+				viewComponent: View,
+			},
+			container
+		);
+	}
+	else {
+		sidebarPanel.open(fetchURL, View);
+	}
+};
+
+export {showSidebar, actions};
+export default showSidebar;
```

Comment:
Noooo, let's not export the same thing twice in two different ways.

Reply:
@wincent should I change it here too then? https://github.com/liferay/liferay-portal/blob/master/modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/Sidebar.js#L94

Just to be sure, do you expect this
```
export {actions};
export default showSidebar;
```
or this
```
export {showSidebar, actions};
```

Reply:
From: https://github.com/liferay/liferay-frontend-guidelines/blob/master/general/file_names.md

| File contents                                 | File name                             |
| --------------------------------------------- | ------------------------------------- |
| A class, `ThingParser`                        | `ThingParser.js`                      |
| A component, `GizmoPicker`                    | `GizmoPicker.js`                      |
| A constructor function, `Blob`                | `Blob.js`                             |
| A function, `concatenateStrings`              | `concatenateStrings.js`               |
| A dictionary of constants, `EVENT_TYPES`      | `EVENT_TYPES.js` (but see note below) |
| Multiple exports, one of which is the default | Based on the default export           |
| Multiple exports, with no default             | Descriptive CamelCase                 |
| Tests for file `src/**/*.js`                  | `test/**/*.js`                        |

And:

> ### How do I pick a "Descriptive CamelCase" name for my file with multiple exports but no default export?
>
> Choosing names is one of [two hard things](https://martinfowler.com/bliki/TwoHardThings.html), so it is not surprising that coming up with a name for a bundle of functionality may be difficult. One way to make the problem simpler is to follow these practices:
>
> -   Prefer small files that each export one thing as opposed to a large bundle that contains many things.
> -   When creating a file with multiple exports is the most practical choice, pick a name that describes the purpose of the contents: if you cannot identify a single purpose, that is a sign that you should find a way to split the file.

So, this one can either be is "multiple exports, one of which is the default" or "multiple exports, with no default". Given that `showSidebar` seems to be the "main" thing exported here, and the file is called `showSidebar.js`, seems like your first option is the right one.

Reply:
Thx!!